operator/internal/lifecycle/v2_node_pools.go

```diff
@@ -40,14 +41,21 @@ func NewV2NodePoolRenderer(mgr ctrl.Manager) *V2NodePoolRenderer {
 // delegating to our particular resource rendering pipeline and filtering out anything that
 // isn't a node pool.
 func (m *V2NodePoolRenderer) Render(ctx context.Context, cluster *redpandav1alpha2.Redpanda) ([]*appsv1.StatefulSet, error) {
-	values := cluster.Spec.ClusterSpec.DeepCopy()
+	if cluster == nil {
```

Comment:
The lack of distinction between optional and by reference in go is one of my least favorite language features.

My rule of thumb is to ask "Is there any case where `variable` could be `nil` that isn't clearly a bug or misuse of the API?" If the answer is "no", don't bother with a check and instead let the code panic.

In this instance, there's no reasonable case for `cluster` to be `nil`. If it is, I'd rather see a panic.

Reply:
ok removed